jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/HttpContent.java

```diff
@@ -19,77 +19,143 @@
+ * of reuse in from a cache).
  * </p>
  */
 public interface HttpContent
 {
+    /**
+     * Get the {@link HttpHeader#CONTENT_TYPE} of this HTTP content.
+     *
+     * @return the content type field, or null if the type of this content is not known.
+     */
     HttpField getContentType();
 
-    String getContentTypeValue();
-
-    String getCharacterEncoding();
-
-    Type getMimeType();
-
+    /**
+     * Get the {@link HttpHeader#CONTENT_ENCODING} of this HTTP content.
+     *
+     * @return the content encoding field, or null if the encoding of this content is not known.
+     */
     HttpField getContentEncoding();
 
-    String getContentEncodingValue();
-
+    /**
+     * Get the {@link HttpHeader#CONTENT_LENGTH} of this HTTP content. The value of the returned field
+     * must always match the value returned by {@link #getContentLengthValue()}.
+     *
+     * @return the content length field, or null if the length of this content is not known.
+     */
     HttpField getContentLength();
 
-    long getContentLengthValue();
+    /**
+     * Get the {@link HttpHeader#LAST_MODIFIED} of this HTTP content. The value of the returned field
+     * must always match the value returned by {@link #getLastModifiedInstant()}.
+     *
+     * @return the last modified field, or null if the last modification time of this content is not known.
+     */
+    HttpField getLastModified();
 
-    Instant getLastModifiedInstant();
+    /**
+     * Get the {@link HttpHeader#ETAG} of this HTTP content.
+     *
+     * @return the ETag, or null if this content has no ETag.
+     */
+    HttpField getETag();
 
-    HttpField getLastModified();
+    /**
+     * Get the character encoding of this HTTP content.
+     *
+     * @return the character encoding, or null if the character encoding of this content is not known.
+     */
+    String getCharacterEncoding();
 
-    String getLastModifiedValue();
+    /**
+     * Get the Mime type of this HTTP content.
+     *
+     * @return the mime type, or null if the mime type of this content is not known.
+     */
+    Type getMimeType();
 
-    HttpField getETag();
+    /**
+     * Get the last modified instant of this resource.
+     *
+     * @return the last modified instant, or null if that instant of this content is not known.
+     * @see #getLastModified()
+     */
+    Instant getLastModifiedInstant();
 
-    String getETagValue();
+    /**
+     * Get the content length of this resource.
+     *
+     * @return the content length of this resource, or -1 if it is not known.
+     * @see #getContentLength()
+     */
+    long getContentLengthValue();
 
     /**
      * Get the {@link Resource} backing this HTTP content.
+     *
      * @return the backing resource.
      */
     Resource getResource();
 
     /**
-     * <p>Write a subset of this HTTP content, to a {@link Content.Sink}.</p>
+     * Asynchronously write a subset of this HTTP content to a {@link Content.Sink}.
```

Comment:
Maybe point out that unlike RBBWwriteTo, this writeTo is not consuming.

Reply:
Good point, done.